### DISCUSSION TITLE
POC C# documentation

### DISCUSSION THREAD

**Discussion by FredBonThermo** [swig/swig#2243] [Show and tell]
Hi everybody,

I'm a developer for **Thermofisher Scientific** company, and my team needs to generate C# with its doxygen documentation.
As it is not available yet in Swig, we decided to implement a POC which is now complete and satisfies our need. 
The POC is available in my dev branch `fbo/indus_doc_csharp`.

This is just a message to inform you about our intention to use our own custom SWIG version in our pipelines.
If you're interested to integrate this branch in a future version, I'm available for any question. 

Best regards

**Comment by wsfulton** [comment]
That sounds interesting and would be a great enhancement to SWIG. I guess this commit contains it all: https://github.com/FredBonThermo/swig/commit/c63e2fc11b725aff7a523c61d14e9adecc148715. Could you please raise a pull request with this so that enhancement can be added into the mainline SWIG. Please remove the numerous whitespace only changes before doing so, coding style guide is at http://swig.org/Doc4.0/Extending.html#Extending_coding_style_guidelines. Looking forward to your contribution.

**Comment by ojwb** [comment]
No obvious progress here for over a year, and meanwhile #2421 has resurrected the older work on doxygen support for SWIG/C# in #1695 so closing this.

It would probably be useful for someone familiar with this area to check if there's anything extra this supports that #2421 doesn't.

**Comment by ojwb** [comment]
Just to note that I've just merged #2835 which implemented support for this (and I guess is actually a continuation of the same work as it's from the same submitter).
